### ISSUE TITLE
C# compiler should not report CS0649 when the fields in question are marked with [DataMember] attribute

### ISSUE THREAD

**Issue by Const-me** [dotnet/roslyn#49616]
**Version Used**: 

Visual studio 2019 16.8.2

**Steps to Reproduce**:

Compile code like this:

```
[DataContract]
sealed class TimeSeriesPoint
{
	[DataMember]
	public float compliance, volume;
}
```

**Expected Behavior**:

Maybe no warnings at all.

Maybe a different message, like "...will always have its default value 0, unless set by deserializing".

**Actual Behavior**:

Warning CS0649: Field 'TimeSeriesPoint.volume' is never assigned to, and will always have its default value 0.

This is incorrect because despite these fields are never assigned to, they won't always have a default value, due to serialization.

**Comment by CyrusNajmabadi** [comment]
This sounds be handled by whoever owns the Data contract api. They sound use a diagnostic suppressor to suppress these warnings.
